[AY1920S1-CS2113T-W17-4] Oof

kahlokkee opened the pull request:
@jasperosy @kenlhc @debbiextan

jinyao-lee commented on src/main/java/oof/CommandParser.java:
```diff
@@ -0,0 +1,138 @@
+                line = line.replaceFirst("help ", "");
+            }
+            return new HelpCommand(line);
+        case "done":
```
Magic numbers. Same goes for those below.

jinyao-lee commented on src/main/java/oof/CommandParser.java:
```diff
@@ -0,0 +1,138 @@
+                throw new OofException("OOPS!!! Please enter valid numbers!");
+            } catch (InputMismatchException e) {
+                throw new OofException("OOPS!!! Please enter a valid number!");
+            }
```
Did not SLAP hard enough; different levels of abstraction here.

jinyao-lee commented on src/main/java/oof/CommandParser.java:
```diff
@@ -0,0 +1,138 @@
+            return new ThresholdCommand(line);
+        default:
+            throw new OofException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
```
This method is way too long, as a result of not enough SLAPping. Work on a way to fix this

jinyao-lee commented on src/main/java/oof/Oof.java:
```diff
@@ -0,0 +1,74 @@
+    public void runTest(String line) throws OofException {
+        Command command = CommandParser.parse(line);
+        command.execute(arr, ui, storage);
+    }
```
Test methods should be in test related classes `XyzTest.java`.

jinyao-lee commented on src/main/java/oof/Reminder.java:
```diff
@@ -0,0 +1,96 @@
+                            count++;
+                        } else {
+                            ui.printUpcomingDeadline(count, taskList.getTask(i));
+                            count++;
```
Duplicated codes. Use guard clauses if need be and shift the codes around.

jinyao-lee commented on src/main/java/oof/Reminder.java:
```diff
@@ -0,0 +1,96 @@
+            if (isNoDeadlineReminded(i, taskList, count)) {
+                ui.printNoDeadlines();
+            }
+        }
```
There's a huge arrowhead here.

Also, there's varying levels of abstraction for this method in general (see lines 34-36 as an example of insufficient SLAPping).

jinyao-lee commented on src/main/java/oof/Reminder.java:
```diff
@@ -0,0 +1,96 @@
+    private static final long MILLISECOND_TO_HOUR = 60 * 60 * 1000;
+
+    /**
+     * Checks Task objects dates to determine if it is due soon.
```
Once this method "checks", what does it do? What is the end result?

If this method is supposed to print to users the list of deadlines, your method name and Javadocs comment should have the relevant keywords.

jinyao-lee commented on src/main/java/oof/Reminder.java:
```diff
@@ -0,0 +1,96 @@
+    }
+
+    /**
+     * Checks if deadline is within the threshold.
```
This method returns a boolean, so what is the end result (when user reads the Javadoc comments)?

In summary: Javadoc comments should be concise, directly to the point of what is the end result of this method.

Same for those below.

jinyao-lee commented on src/main/java/oof/Storage.java:
```diff
@@ -0,0 +1,198 @@
+     * @return commandList  ArrayList of available commands.
+     * @throws OofException if manual cannot be retrieved from file path.
+     */
+    public ArrayList<String> readManual() throws OofException {
```
The name of the method doesn't correspond to the Javadoc comment. The end result is to return something, so the name of the method should start with the keyword that correspond to returning the list of commands available to user.

Same goes for those below.

jinyao-lee commented on src/main/java/oof/Storage.java:
```diff
@@ -0,0 +1,198 @@
+            out.write(updateThreshold);
+            out.close();
+        } catch (IOException e) {
+            System.out.println(e + ", thus please try inputting other things.");
```
- `", thus please try inputting other things."` is repeated below as well. You can make it a constant class level variable for this. Same for others.

jinyao-lee commented on src/main/java/oof/Storage.java:
```diff
@@ -0,0 +1,198 @@
+        }
+        reader.close();
+        return arr;
+    }
```
Avoid long methods. Same goes for those below.

jinyao-lee commented on src/main/java/oof/Ui.java:
```diff
@@ -0,0 +1,757 @@
+            System.out.println("\tNow you have " + size + " task in the list.");
+        }
+        printLine();
+    }
```
Violates SRP here, where you are dealing with task related contents here as well.

jinyao-lee commented on src/main/java/oof/Ui.java:
```diff
@@ -0,0 +1,757 @@
+                return input;
+            }
+        }
+    }
```
Weird logic here. Method is supposed to print something but here you are asking for user input?

jinyao-lee commented on src/main/java/oof/Ui.java:
```diff
@@ -0,0 +1,757 @@
+     * @param largestColSize Size of the largest column in the View Week output.
+     */
+    public void printViewWeek(ArrayList<ArrayList<String[]>> tasks, Date startDate, int largestTaskSize,
+                              int largestColSize) {
```
https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#layout

We are following the OSS coding standards for this mod, so indentation issue here, where wrapped lines should have 8 spaces instead of using the default intellij wrapping.

jinyao-lee commented on src/main/java/oof/Ui.java:
```diff
@@ -0,0 +1,757 @@
+            for (int day = 0; day < DAYS_IN_WEEK; day++) {
+                String dayString = calendarDates.get(dayIndex + day).trim();
+                if (dayString.equals("") || calendar.get(Integer.parseInt(dayString)).size() <= row) {
+                    System.out.print("                     |");
```
You can find a way to print `x` number of spaces instead of hard-coding it.

This creates problems when you are running test cases.

jinyao-lee commented on src/main/java/oof/command/CalendarCommand.java:
```diff
@@ -0,0 +1,155 @@
+                // 0-based indexing to 1-based indexing
+                this.month = calendar.get(Calendar.MONTH) + 1;
+                this.year = calendar.get(Calendar.YEAR);
+            }
```
I (as the reader) don't get what this code is trying to achieve. 

I.e.: Write a comment for non-trivial codes

jinyao-lee commented on src/main/java/oof/command/CalendarCommand.java:
```diff
@@ -0,0 +1,155 @@
+     */
+    class SortByTime implements Comparator<String[]> {
+        @Override
+        public int compare(String[] a, String[] b) {
```
Missing Javadocs comment. All public and non-trivial private methods should have one.

jinyao-lee commented on src/main/java/oof/command/RecurringCommand.java:
```diff
@@ -0,0 +1,177 @@
+    private static final int WEEKLY = 2;
+    private static final int MONTHLY = 3;
+    private static final int YEARLY = 4;
+    private static final int WEEK = 7;
```
`WEEK` doesn't explain to the reader anything. Is it the number of days in a week?

jinyao-lee reviewed (commented):
Good effort in your codes!

Note that whatever comments that I've made extends to the rest of your codes as well.

Things to note:
- Insufficient SLAPping in some parts of your code.
- Variable names need to make sense. For example, `private static final int WEEK = 7` does not tell the reader much about the purpose of this WEEK. Don't make the reader guess the purpose of it, for it always go wrong.

debbiextan committed:
Implement Time Period functionality for Tracker View Command.
Update User Guide with Time Period functionality for Tracker View Command.

jasperosy committed:
Merge pull request #179 from debbiextan/master

Implement Time Period functionality for Tracker Command and refactor Tracker Command related code

Kenlhc committed:
Testing code

Kenlhc committed:
Merge https://github.com/AY1920S1-CS2113T-W17-4/main

Kenlhc committed:
Implement method to suggest tasks to do during free time.

Kenlhc committed:
Cleaned up code based on comments.

Kenlhc committed:
Fixed checkstyle errors.

kahlokkee committed:
Merge pull request #180 from Kenlhc/master

Implement method to suggest deadlines to complete during free time

jasperosy committed:
Update architecture and manual testing sections

jasperosy committed:
Include architecture diagram

kahlokkee committed:
Add Exceptions and Testing code

kahlokkee committed:
Fix Checkstyle errors

kahlokkee committed:
Merge pull request #181 from KahLokKee/master

Add Exceptions and Testing code

kahlokkee committed:
Update shadowJar version

Kenlhc committed:
Added Junit Tests, updated threshold output and manual.

Kenlhc committed:
Fixed Junit Tests.

jasperosy committed:
Update DeveloperGuide.adoc

jasperosy committed:
Merge branch 'master' into master

jasperosy committed:
Merge pull request #184 from Kenlhc/master

Added Junit Tests, updated threshold output and manual.

jasperosy committed:
Delete ArchitectureDiagram.png

jasperosy committed:
Add files via upload

jasperosy committed:
Update DeveloperGuide.adoc

kahlokkee committed:
Add support for Lesson class in Calendar

Kenlhc committed:
Fixed bugs in threshold command.

kahlokkee committed:
Fix checkstyle error

jasperosy committed:
Merge pull request #185 from KahLokKee/master

Add support for Lesson class in Calendar

jasperosy committed:
Merge branch 'master' into master

jasperosy committed:
Merge pull request #186 from Kenlhc/master

Fixed bugs in threshold command.

jasperosy committed:
Add lesson support for viewweek command

kahlokkee committed:
Merge pull request #187 from jasperosy/master

Add lesson support for viewweek command

kahlokkee committed:
Add Storage support for reading .jar files

kahlokkee committed:
Fix checkstyle error

Kenlhc committed:
Added support for lessons in FreeCommand.

Kenlhc committed:
Merge remote-tracking branch 'refs/remotes/origin/master'

kahlokkee committed:
Merge pull request #189 from Kenlhc/master

Added support for lessons in FreeCommand

kahlokkee committed:
Merge remote-tracking branch 'upstream/master'

kahlokkee committed:
Fix known bugs

kahlokkee committed:
Fix more bugs.

Kenlhc committed:
Merge pull request #188 from KahLokKee/master

Add Storage support for reading .jar files and bug fixes

jasperosy committed:
Fix viewweek bug

jasperosy committed:
Edit code that I have missed out

jasperosy committed:
Fix bug for week crossing over to next year

kahlokkee committed:
Merge pull request #190 from jasperosy/master

Fix viewweek bug

debbiextan committed:
Add Summary, Schedule, Tracker and Help Command test code.
Update manual.txt
Make minor edits to Tracker isNotStarted method.

debbiextan committed:
Merge remote-tracking branch 'upstream/master'

debbiextan committed:
Fix checkstyle errors.

kahlokkee committed:
Minor UI changes for Calendar Command

kahlokkee committed:
Fix command parsing bugs

kahlokkee committed:
Increase maximum description length for Semester and Module name

jasperosy committed:
Merge pull request #192 from KahLokKee/master

Minor UI changes for Calendar Command and additional bug fixes

jasperosy committed:
Merge branch 'master' into master

debbiextan committed:
Merge remote-tracking branch 'upstream/master'

debbiextan committed:
Handle negative cases of Schedule and Summary Command.
Refactor scheduleByDate method in Schedule Command.
Add JUnit test for Delete Command.
Fix JUnit test for Schedule and Summary Command.

debbiextan committed:
Merge remote-tracking branch 'origin/master'

debbiextan committed:
Remove Summary Command Test.
Change Schedule Command Test date.

debbiextan committed:
Remove Summary Command Test.
Change Schedule Command Test date.
Remove MANIFEST file (from previous commit).

jasperosy committed:
Delete RecurringCommandSequenceDiagram.png

jasperosy committed:
Delete ViewWeekSequenceDiagram.png

jasperosy committed:
Add files via upload

jasperosy committed:
Update README.adoc

jasperosy committed:
Update DeveloperGuide.adoc

jasperosy committed:
Delete ViewWeekSequenceDiagram.png

jasperosy committed:
Delete RecurringCommandSequenceDiagram.png

jasperosy committed:
Add files via upload

debbiextan committed:
Fix checkstyle errors.

kahlokkee committed:
Upload Semester and Task Class Diagram

debbiextan committed:
Fix checkstyle errors.
Refactor Delete Task Command Test execute correctIndex deleteTask method.

jasperosy committed:
Delete ViewWeekCommandSequenceDiagram.png

jasperosy committed:
Delete RecurringCommandSequenceDiagram.png

jasperosy committed:
Add files via upload

debbiextan committed:
Merge remote-tracking branch 'upstream/master'

kahlokkee committed:
Update SemesterTaskClassDiagram.png

kahlokkee committed:
Move files into corresponding packages

jasperosy committed:
Merge pull request #194 from KahLokKee/master

Move files into corresponding packages

jasperosy committed:
Delete ViewWeekCommandSequenceDiagram.png

jasperosy committed:
Add files via upload

jasperosy committed:
Delete ArchitectureDiagram.png

jasperosy committed:
Add files via upload

jasperosy committed:
Update architecture diagram and content

debbiextan committed:
Refactor TrackerCommandTest.
Implement Delete and List functions for Tracker Command.
Update manual.txt and UG with new Tracker Command syntax.

debbiextan committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	src/main/java/oof/logic/command/organization/exceptions/ScheduleEmptyException.java
#	src/main/java/oof/logic/command/productivity/ScheduleCommand.java
#	src/main/java/oof/logic/command/productivity/SummaryCommand.java

debbiextan committed:
Refactor TrackerCommandTest.
Implement Delete and List functions for Tracker Command.
Update manual.txt and UG with new Tracker Command syntax.
Fix merge conflict.

Kenlhc committed:
Delete FreeCommandActivityDiagram.png

Kenlhc committed:
Delete FreeTimeDisplay.png

Kenlhc committed:
Add files via upload

Kenlhc committed:
Added sequence diagram and details for FreeCommand

Kenlhc committed:
Merge branch 'master' into master

Kenlhc committed:
Merge pull request #191 from debbiextan/master

Add JUnit test code and other minor changes

jasperosy committed:
Delete ArchitectureDiagram.png

jasperosy committed:
Add files via upload

Kenlhc committed:
Bug fixes. Minor name changes.

Kenlhc committed:
Fixed Travis Errors.

Kenlhc committed:
Merge pull request #195 from Kenlhc/master

Bug fixes for ScheduleCommand. Minor name changes.

jasperosy committed:
Create UsingGradle.adoc

jasperosy committed:
Create UsingTravis.adoc

jasperosy committed:
Create UsingCoveralls.adoc

jasperosy committed:
Create UsingNetlify.adoc

jasperosy committed:
Update UsingTravis.adoc

jasperosy committed:
Update UsingCoveralls.adoc

jasperosy committed:
Update introduction, documentation, testing and dev ops sections

Kenlhc committed:
Update UserGuide.adoc

jasperosy committed:
Update DeveloperGuide.adoc

jasperosy committed:
Add files via upload

jasperosy committed:
Add files via upload

Kenlhc committed:
Delete FreeTimeExample.png

Kenlhc committed:
Delete Threshold.png

Kenlhc committed:
Add files via upload

jasperosy committed:
Update UsingNetlify.adoc

jasperosy committed:
Update UsingCoveralls.adoc

jasperosy committed:
Update UsingCoveralls.adoc

jasperosy committed:
Update UsingNetlify.adoc

jasperosy committed:
Update DeveloperGuide.adoc

kahlokkee committed:
Add files via upload

jasperosy committed:
Update DeveloperGuide.adoc

jasperosy committed:
Update DeveloperGuide.adoc

jasperosy committed:
Update UsingTravis.adoc

jasperosy committed:
Update UserGuide.adoc

kahlokkee committed:
Update DG for Model Component and add additional tests for semester package

jasperosy committed:
Merge pull request #200 from KahLokKee/master

Update DG for Model Component and add additional testing code

jasperosy committed:
Delete RecurringCommandSequenceDiagram.png

jasperosy committed:
Add files via upload

Kenlhc committed:
Update DeveloperGuide.adoc

debbiextan committed:
Add Help, Schedule, Summary and Tracker DG diagrams.

kahlokkee committed:
Update UG and DG

kahlokkee committed:
Rename Calendar.png

jasperosy committed:
Update instructions for setting up

jasperosy committed:
Update UserGuide.adoc

debbiextan committed:
Add Help, Schedule, Summary and Tracker DG diagrams.
Update UG with Tracker Command.

jasperosy committed:
Merge branch 'master' into master

jasperosy committed:
Merge pull request #201 from KahLokKee/master

Update UG

debbiextan committed:
Merge remote-tracking branch 'upstream/master'

debbiextan committed:
Update Help Command warning admonitions in UG.
Add manual testing for Tracker commands.

jasperosy committed:
include Jansi library to remove batch file dependency

jasperosy committed:
Merge pull request #202 from debbiextan/master

DG Update

jasperosy committed:
Merge remote-tracking branch 'upstream/master'

jasperosy committed:
Change error messages for trackercommand

jasperosy committed:
update tracker command test

jasperosy committed:
Merge pull request #203 from jasperosy/master

Include Jansi library to remove batch file dependency

jasperosy committed:
Update README.adoc

jasperosy committed:
Update UserGuide.adoc

jasperosy committed:
Update UserGuide.adoc

jasperosy committed:
Update UserGuide.adoc

jasperosy committed:
Update DeveloperGuide.adoc

jasperosy committed:
Update DeveloperGuide.adoc

jasperosy committed:
Update DeveloperGuide.adoc

Kenlhc committed:
Update DeveloperGuide.adoc

Kenlhc committed:
Update DeveloperGuide.adoc

Kenlhc committed:
Add files via upload

kahlokkee committed:
Update DG with Manual Testing instructions for Semester and Module

kahlokkee committed:
Standardize labelling conventions for DG

kahlokkee committed:
Edit AddSemesterCommandTest to match AddSemesterCommand changes

jasperosy committed:
Merge pull request #204 from KahLokKee/master

Update DG with Manual Testing instructions for Semester and Module

jasperosy committed:
Delete Help.png

jasperosy committed:
Update help command image

jasperosy committed:
Fix grammatical errors and spelling mistakes

jasperosy committed:
Delete UserGuide.pdf

jasperosy committed:
Render adoc to pdf for UG

Kenlhc committed:
Update DeveloperGuide.adoc

Kenlhc committed:
Delete UiClassDiagram.png

Kenlhc committed:
Add files via upload

kahlokkee committed:
Update DG with Manual Testing for Lesson and Tasks and bug fix for AddLessonCommand

jasperosy committed:
Merge pull request #205 from KahLokKee/master

Update DG with Manual Testing for Lesson and Tasks and bug fix

debbiextan committed:
Add Logic Component to DG.

debbiextan committed:
Merge remote-tracking branch 'upstream/master'

debbiextan committed:
Add relevant Tracker images to UG and DG.

jasperosy committed:
Merge pull request #206 from debbiextan/master

UG and DG Update

jasperosy committed:
Format Tracker command images' layout

jasperosy committed:
Update DeveloperGuide.adoc

kahlokkee committed:
Update UserGuide.adoc

kahlokkee committed:
Update UserGuide.adoc

kahlokkee committed:
Update DeveloperGuide.adoc

kahlokkee committed:
Update DeveloperGuide.adoc

kahlokkee committed:
Update DeveloperGuide.adoc

kahlokkee committed:
Update UserGuide.adoc

kahlokkee committed:
Update UserGuide.adoc

jasperosy committed:
Final UG update

debbiextan committed:
Add relevant Logic component diagrams via upload.

debbiextan committed:
Merge remote-tracking branch 'upstream/master'

jasperosy committed:
Merge pull request #207 from debbiextan/master

Update Logic component

kahlokkee committed:
Move EmptyListExceptions to commons

jasperosy committed:
Merge pull request #208 from KahLokKee/master

Move EmptyListExceptions to commons

Kenlhc committed:
Delete StorageClassDiagram.png

Kenlhc committed:
Add files via upload

Kenlhc committed:
Update DeveloperGuide.adoc

debbiextan committed:
Fix Logic component class diagrams.

jasperosy committed:
Merge pull request #209 from debbiextan/master

Fix Logic component class diagrams.

Kenlhc committed:
Update DeveloperGuide.adoc

Kenlhc committed:
Delete FreeCommandSequenceDiagram.png

Kenlhc committed:
Add files via upload

Kenlhc committed:
Delete FreeCommand Sequence Diagram.png

Kenlhc committed:
Add files via upload

Kenlhc committed:
Update DeveloperGuide.adoc

kahlokkee committed:
Delete CalendarSequenceDiagram.png

kahlokkee committed:
Add files via upload

kahlokkee committed:
Delete CalendarSequenceDiagram.png

kahlokkee committed:
Add files via upload

debbiextan committed:
Edit Step 1 of Tracker feature in DG.

debbiextan committed:
Fix grammar for Step 1 in tracker command.

jasperosy committed:
Delete ViewWeekSequenceDiagram.png

jasperosy committed:
Add files via upload

jasperosy committed:
Add files via upload

Kenlhc committed:
Delete FreeCommandSequenceDiagram.png

Kenlhc committed:
Add files via upload

jasperosy committed:
Merge branch 'master' into master

jasperosy committed:
Merge pull request #210 from debbiextan/master

Edit Task Tracker Implementation in DG.

debbiextan committed:
Handle index out of bounds error and number format exception in TrackerCommand.

debbiextan committed:
Handle index out of bounds error and number format exception in TrackerCommand.
Add author tags.

debbiextan committed:
Merge remote-tracking branch 'upstream/master'

kahlokkee committed:
Update UG and bug fixes

debbiextan committed:
Handle index out of bounds error and number format exception in TrackerCommand.
Add author tags.

jasperosy committed:
Merge pull request #211 from debbiextan/master

Fix bugs in Tracker Command and add author tags

jasperosy committed:
Merge branch 'master' into master

kahlokkee committed:
Fix date parsing bugs

jasperosy committed:
Merge pull request #212 from KahLokKee/master

Update UG and bug fixes

Kenlhc committed:
Update UserGuide.adoc

Kenlhc committed:
Delete Threshold.png

Kenlhc committed:
Add files via upload

jasperosy committed:
Delete UserGuide.pdf

jasperosy committed:
Add files via upload

jasperosy committed:
Update DeveloperGuide.adoc

kahlokkee committed:
Update DG

kahlokkee committed:
Merge remote-tracking branch 'upstream/master'

jasperosy committed:
Merge pull request #213 from KahLokKee/master

Update DG

jasperosy committed:
Correct grammatical errors

Kenlhc committed:
Update DeveloperGuide.adoc

Kenlhc committed:
Update DeveloperGuide.adoc

kahlokkee committed:
Delete LogicOrganizationClassDiagram.png

kahlokkee committed:
Add files via upload

Kenlhc committed:
Delete StorageClassDiagram.png

Kenlhc committed:
Add files via upload

kahlokkee committed:
Update DG and added support for Lesson in Summary and Schedule Command

jasperosy committed:
Merge pull request #214 from KahLokKee/master

Update DG and added support for Lesson in Summary and Schedule Command

jasperosy committed:
Delete DeveloperGuide.pdf

jasperosy committed:
Delete UserGuide.pdf

jasperosy committed:
Add files via upload

jasperosy committed:
populate output.txt in resources

debbiextan committed:
Update DG Manual Testing segment for Task Tracker feature.

kahlokkee committed:
Improve Code Quality

kahlokkee committed:
Merge branch 'master' into master

jasperosy committed:
Merge branch 'master' into master

kahlokkee committed:
Fix checkstyle error

jasperosy committed:
Merge pull request #216 from debbiextan/master

Update DG Manual Testing segment for Task Tracker feature.

jasperosy committed:
Merge remote-tracking branch 'upstream/master'

jasperosy committed:
Delete DeveloperGuide.pdf

jasperosy committed:
Add files via upload

jasperosy committed:
Merge branch 'master' into master

jasperosy committed:
Merge pull request #217 from KahLokKee/master

Improve Code Quality

jasperosy committed:
Merge remote-tracking branch 'upstream/master'

jasperosy committed:
revert output.txt

jasperosy committed:
Merge pull request #215 from jasperosy/master

populate output.txt in resources
